docs/cloud/execution/fargate_task_environment.md

```diff
@@ -0,0 +1,178 @@
+
+Accepted kwargs for [`register_task_definition`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.register_task_definition):
+
+```
+family                      string
+taskRoleArn                 string
+executionRoleArn            string
+networkMode                 string
+containerDefinitions        list
+volumes                     list
+placementConstraints        list
+requiresCompatibilities     list
+cpu                         string
+memory                      string
+tags                        list
+pidMode                     string
+ipcMode                     string
+proxyConfiguration          dict
+inferenceAccelerators       list
+```
+
+Accepted kwargs for [`run_task`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.run_task):
+
+```
+cluster                     string
+taskDefinition              string
+count                       integer
+startedBy                   string
+group                       string
+placementConstraints        list
+placementStrategy           list
+platformVersion             string
+networkConfiguration        dict
+tags                        list
+enableECSManagedTags        boolean
+propagateTags               string
+```
+
+All of these kwargs will be loaded and stored upon initialization of the Environment. It will never be sent to Prefect Cloud and will only exist inside your Flow's Docker storage.
+
+:::tip Task IAM Roles
+Users have seen great performance in using [Task IAM Roles](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-iam-roles.html) for their Flow execution.
+:::
+
+#### Setup
+
+The Fargate Task Environment setup step is responsible for registering the Fargate Task if it does not already exist. First it checks for the existence of a task definition based on the `family` that was provided at initialization of this Environment. If the task definition is not found then it is created. This means that if a Flow is run multiple times the task definition will only need to be created once.
+
+#### Execute
+
+Create a new Fargate Task with the configuration provided at initialization of this Environment. That task is responsible for running the Flow.
+
+#### Task Spec Configuration
+
+There are a few caveats to using the Fargate Task Environment that revolve around the provided boto3 kwargs. In the `containerDefinitions` that you provide the **first container** listed will be the container that is used to run the Flow. This means that the first container will always be overridden during the `setup` step of this Environment.
```

Comment:
```suggestion
There are a few caveats to using the Fargate Task Environment that revolve around the provided boto3 kwargs. In the `containerDefinitions` that you provide, the **first container** listed will be the container that is used to run the Flow. This means that the first container will always be overridden during the `setup` step of this Environment.
```

Potentially I'm not understanding this, but the first container listed will be the one used to run the Flow but the first container will always be overridden? Those feel like opposing statements